components/mailgun/mailgun.app.js

```diff
@@ -0,0 +1,153 @@
+      [402, 429, 500, 502, 503, 504].includes(statusCode);
+      //TODO: Remove this comment. For code review purposes, I am linking to Mailgun's HTTP status code returned: https://documentation.mailgun.com/en/latest/api-intro.html?highlight=502#status-codes Errro 408, common in other Pipedream retries documentedly not returned by Mailgun, so it was not included.
+    },
+    async _withRetries(apiCall, allow404 = false) {
+      const retryOpts = {
+        retries: 5,
+        factor: 2,
+      };
+      return retry(async (bail) => {
+        try {
+          return await apiCall();
+        } catch (err) {
+          const statusCode = get(err, ["response", "status"]);
+          if (!this._isRetriableStatusCode(statusCode)) {
+            bail(`
+              Unexpected error (status code: ${statusCode}):
+              ${JSON.stringify(err.response)}
+            `);
+          }
+          console.warn(`Temporary error: ${err.message}`);
+          throw err;
+        }
+      }, retryOpts);
+    },
+    async getDomains(skip) {
+      return await this._withRetries(() =>
+        this._makeRequest({
+          path: `/v3/domains`,
+          params: {
+            skip,
+          },
+        })
+      );
+    },
+    async getWebhookDetails(mailgunDomain, webhookName) {
+      const webhooks = await this._withRetries(() =>
+        this._makeRequest({
+          path: `/v3/domains/${mailgunDomain}/webhooks`,
+          method: "get",
+        })
+      );
+      return get(webhooks, `webhooks.${webhookName}`);
+    },
+    async getAllDomains() {
+      let domains_set;
+      const results = [];
+      let skip = 0;
+      do {
+        domains_set = await this.getDomains(skip);
+        domains_set.items.forEach((domain) => {
+          results.push({
+            name: domain.name,
+          });
+        });
+        skip = skip + 100;
+      } while (domains_set.items.length);
+      return results;
+    },
+    async createWebhook(mailgunDomain, webhookName, webhookUrl) {
+      return await this._withRetries(() =>
+        this._makeRequest({
+          method: "post",
+          path: `v3/domains/${mailgunDomain}/webhooks`,
+          params: {
+            domain: mailgunDomain,
+            url: webhookUrl,
+            id: webhookName,
+          },
+        })
+      );
+    },
+    async updateWebhook(mailgunDomain, webhookName, webhookUrls) {
+      const FormData = require("form-data");
+      data = new FormData();
+      data.append("domain", mailgunDomain);
+      data.append("webhookname", webhookName);
+      webhookUrls.forEach((url) => {
+        data.append("url", url);
+      });
+      const opts = {
+        method: "put",
+        headers: {
+          "Content-Type": `multipart/form-data; boundary=${data._boundary}`,
+        },
+        path: `v3/domains/${mailgunDomain}/webhooks/${webhookName}`,
+        data,
+      };
+      return await this._withRetries(() => this._makeRequest(opts));
```

Comment:
Any reason why we're not using the [official Mailgun SDK](https://github.com/mailgun/mailgun-js)?
Using an official SDK should is preferrable unless there's a good reason not to (e.g. flexibility).

Reply:
Ah, sorry. I see there are a couple of actions already using mailgun SDK, but the Pipedream code to create an action `"Run Node.js code with Mailgun"` is using axios and hitting the API with HTTP request, and some how I based off from that. Should I switch to the SDK then?